### PR TITLE
[SELS-TASK][FE/BE] Admin Userlist Page and Functions

### PR DESCRIPTION
**Pull request by michaelsilverio**
### [Asana Link]
[link here](https://app.asana.com/0/1201713838250812/1201726038594833/f)

### [Commands]
```
# Made changes to the database fields | new database seeders data
php artisan migrate:fresh --seed

# Deploying application
php artisan serve
npm run watch
```

### [Pre-conditions]
After running `npm run watch` and `php artisan serve`. Go to the url being generated on artisan serve

// Sign In Details
ADMIN | email: admin@admin.com | password: admin
### [Expected Output]
Details:

- When you first sign in as Admin, you will be redirected to the categories list. When you click the navigation menu link e.g. Userlist, if you try to reload the page, it will not go back to the categories list. When you click again the link to categories list the categories list page willl be the current page and when you reload the page it will still be on the categories list page.
- On the Userlist page, you can update the role of a user
- If the user is admin then you can only update that user with a new `user` role, if the user is `user` then you can only update the user to new admin role.
- Added the pagination. You can go to the next page or go back to previous page
- When you update a user on the next page it will not go back to the previous page.

#### Word page issue:
- There was a problem with the word page. When you update a word list row it would work on first update but when you update again, it will not update the word list row component. This is now resolved by adding the function to re-initialize the state.

### [Screenshots]
image

image

### CODE REVIEW

File: resources/js/components/layouts/Header.js [hunk 1]
```diff
@@ -34,46 +35,74 @@ const Header = (props) => {
...
+    if (uriName !== undefined) {
+      if (uriName === "/dashboard") {
+        setURI(uriName);
+      }
+      if (uriName === "/categories") {
+        setURI(uriName);
+      }
+      if (uriName === "/user_lists") {
+        setURI(uriName);
+      }
```
Comment (AubreyV) [inline resources/js/components/layouts/Header.js]: you can use switch-case here instead of having multiple ifs

File: public/js/app.js [hunk 2]
```diff
@@ -39043,16 +39043,15 @@ var Header = function Header(props) {
     var uriName = props.cookies.get("activePage");
 
     if (uriName !== undefined) {
-      if (uriName === "/dashboard") {
-        setURI(uriName);
-      }
+      switch (uriName) {
+        case "/dashboard":
+          setURI(uriName);
```
Comment (AubreyV) [inline public/js/app.js:39048]: put `break` after every `case`
https://www.w3schools.com/js/js_switch.asp

File: resources/js/components/layouts/Header.js [hunk 3]
```diff
@@ -38,14 +38,13 @@ const Header = (props) => {
...
-      }
-      if (uriName === "/categories") {
-        setURI(uriName);
-      }
-      if (uriName === "/user_lists") {
-        setURI(uriName);
+      switch (uriName) {
+        case "/dashboard":
+          setURI(uriName);
+        case "/categories":
```
Comment (AubreyV) [inline resources/js/components/layouts/Header.js:45]: here also https://www.w3schools.com/js/js_switch.asp

File: public/js/app.js (hunk 2 above)
Comment (michaelsilverio) [reply public/js/app.js:39048]: This is resolved ma'am.

File: resources/js/components/layouts/Header.js (hunk 3 above)
Comment (michaelsilverio) [reply resources/js/components/layouts/Header.js:45]: This is resolved ma'am.

**Pull request merged**